### DISCUSSION TITLE
LAB 05.02 Task 02

### DISCUSSION THREAD

**Discussion by camilo1109** [rramosp/ai4eng.v1#68] [05 - GESTIÓN DE MODELOS]
Buenas tardes, quisiera sabes cual es el error que tengo en el diseño de esta actividad. adjunto foto de la definición de la función y a darle submit me sale siempre algo así.

def fit_and_predict(estimator, X, y):
    assert len(X)==len(y), "X and y must have the same length"
    permutacion = np.random.permutation(len(X))
    part1 = permutacion[:int(len(X)/2)]
    print(part1)
    part2 = permutacion[int(len(X)/2):]
    print(part2)

    estimator.fit(X[part1],y[part1])
    
    predictions = estimator.predict(X[part2])
    
    return estimator, predictions

task_02 submitted. your grade is 0
----- grader message -------
incorrect: testing with 100 random calls
predictions do not match
X=
[[ 0.29950508  0.57686543]
 [ 0.89401489 -0.83923648]
 [-0.83425169  0.7190602 ]
 [-0.57183854  0.5076414 ]
 [ 1.76604827 -0.21549086]
 [-0.10682585  0.84380633]
 [ 0.82427313  0.39889671]
 [-1.29618106  0.15577904]
 [-0.02902751  0.54414491]
 [ 0.46122704  0.85453117]
 [ 0.73185698 -0.35982455]
 [ 2.16772136  0.1097706 ]
 [ 2.06117715 -0.58586438]
 [-1.33286404  0.27293442]
 [ 0.01487071  0.98919099]
 [ 0.676948    0.45305159]
 [ 0.42528351  0.08380599]
 [ 1.30362083 -0.54581195]
 [ 0.82483846  1.03178679]
 [ 0.31408464 -0.20784327]
 [ 0.96657033  0.38888887]
 [ 2.12815408  0.72558769]]

y=
[1 1 0 0 1 0 0 0 1 0 1 1 1 0 0 0 1 1 0 1 0 1]

estimator
DecisionTreeClassifier(ccp_alpha=0.0, class_weight=None, criterion='gini',
                       max_depth=3, max_features=None, max_leaf_nodes=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, presort='deprecated',
                       random_state=0, splitter='best')

expected predictions
[1 1 0 0 1 1 1 0 1 1 0]

but obtained predictions
[1 0 0 1 1 0 1 1 1 0 1] 

Muchísimas gracias!

**Comment by camilo1109** [comment]
ya di con el punto, el caso es que no había leído bien que las particiones era no-aleatorias.

**Comment by rramosp** [comment]
perfecto camilo! me alegro que hayas podido resolverlo!!

El dom, 24 oct 2021 a las 16:35, camilo1109 ***@***.***>)
escribió:

> ya di con el punto, el caso es que no había leído bien que las particiones
> era no-aleatorias.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/68#discussioncomment-1528813>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIJXXCFHL3YS27PZPWDUIR33HANCNFSM5GTYQGSA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.
